macro/src/span.rs

```diff
@@ -0,0 +1,93 @@
+use proc_macro::Span;
+use std::collections::hash_map::DefaultHasher;
+use std::env;
+use std::hash::{Hash, Hasher};
+
+fn seed() -> impl Hash {
+    env::var_os("CONST_RANDOM_SEED")
+}
+
+pub(crate) fn gen_random<T: Random>() -> T {
+    Random::random()
+}
+
+pub(crate) trait Random {
+    fn random() -> Self;
+}
+
+impl Random for u8 {
+    fn random() -> Self {
+        u64::random() as u8
+    }
+}
+
+impl Random for u16 {
+    fn random() -> Self {
+        u64::random() as u16
+    }
+}
+
+impl Random for u32 {
+    fn random() -> Self {
+        u64::random() as u32
+    }
+}
+
+impl Random for u64 {
+    fn random() -> Self {
+        let span = Span::call_site();
+        let mut hasher = DefaultHasher::new();
+        seed().hash(&mut hasher);
```

Comment:
Is this guaranteed to be stable accross compiler changes?

Reply:
It isn't relevant because rustc's output is not going to be the same across different compilers anyway, regardless of whether this is. This PR is about taking a crate that uses const_random and compiling it to a deterministic cacheable artifact. If you compile the same crate with a different compiler, you will get a different artifact, even if const_random is not used.